src/settings/mod.rs

```diff
@@ -1,28 +1,72 @@
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, you can obtain one at https://mozilla.org/MPL/2.0/.
-
 //! Application settings.
 
-use std::env;
-
 use config::{Config, ConfigError, Environment, File};
+use serde::de::{Deserialize, Deserializer, Error, Unexpected};
+use std::{
+    env,
+    fmt::{self, Display},
+};
 
-use deserialize;
+use duration::Duration;
 use logging::MozlogLogger;
 use serialize;
+use validate;
 
 #[cfg(test)]
 mod test;
 
+macro_rules! deserialize {
+    ($(($type:ident, $validator:ident, $expected:expr)),+) => ($(
+        #[derive(Clone, Debug, Default, Serialize, PartialEq)]
+        pub struct $type(pub String);
+
+        impl Display for $type {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                f.write_str(&self.0)
+            }
+        }
+
+        impl<'d> Deserialize<'d> for $type {
+            fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+            where
+                D: Deserializer<'d>,
+            {
+                let value: String = Deserialize::deserialize(deserializer)?;
+                if validate::$validator(&value) {
```

Comment:
Maybe here I should be more generic and instead of being bound to the `validate` module I should just receive a `Fn(&str) -> bool` as `$validator` instead...